test/integration/user_commands.cc

```diff
@@ -1031,22 +1031,350 @@ TEST_F(UserCommandsTest, GZ_UTILS_TEST_ENABLED_ONLY_ON_LINUX(Light))
   const std::string lightTopic = "/world/lights_command/light_config";
 
   msgs::Light lightMsg;
-  lightMsg.set_name("spot");
   gz::msgs::Set(lightMsg.mutable_diffuse(),
     gz::math::Color(1.0f, 1.0f, 1.0f, 1.0f));
+  gz::msgs::Set(lightMsg.mutable_pose()->mutable_position(),
+    gz::math::Vector3d(1.0f, 0.0f, 0.0f));
 
-  // Publish light config
+  // Publish light config without name
   auto pub = node.Advertise<msgs::Light>(lightTopic);
   pub.Publish(lightMsg);
 
   server.Run(true, 100, false);
   // Sleep for a small duration to allow Run thread to start
   GZ_SLEEP_MS(10);
 
+  // add name
+  lightMsg.set_name("spot");
+
+  // Publish light config
+  pub.Publish(lightMsg);
+
+  server.Run(true, 100, false);
+  // Sleep for a small duration to allow Run thread to start
+  GZ_SLEEP_MS(10);
+
   EXPECT_EQ(math::Color(1.0f, 1.0f, 1.0f, 1.0f),
     spotLightComp->Data().Diffuse());
 }
 
+/////////////////////////////////////////////////
+TEST_F(UserCommandsTest, GZ_UTILS_TEST_ENABLED_ONLY_ON_LINUX(LightAll))
+{
+  // Start server
+  ServerConfig serverConfig;
+  const auto sdfFile = gz::common::joinPaths(
+    std::string(PROJECT_SOURCE_PATH), "test", "worlds",
+      "lights_render_all.sdf");
+  serverConfig.SetSdfFile(sdfFile);
+
+  Server server(serverConfig);
+  EXPECT_FALSE(server.Running());
+  EXPECT_FALSE(*server.Running(0));
+
+  // Create a system just to get the ECM
+  EntityComponentManager *ecm{nullptr};
+  test::Relay testSystem;
+  testSystem.OnPreUpdate([&](const sim::UpdateInfo &,
+                             sim::EntityComponentManager &_ecm)
+      {
+        ecm = &_ecm;
+      });
+
+  server.AddSystem(testSystem.systemPtr);
+
+  // Run server and check we have the ECM
+  EXPECT_EQ(nullptr, ecm);
+  server.Run(true, 1, false);
+  EXPECT_NE(nullptr, ecm);
+
+  msgs::Light req;
+  msgs::Boolean res;
+  transport::Node node;
+  bool result;
+  unsigned int timeout = 1000;
+  std::string service{"/world/lights_command/light_config"};
```

Comment:
include `<string>`

Reply:
Hmmmm, not sure what you are suggesting with this one? I'm following how it is done in all other tests?:

https://github.com/gazebosim/gz-sim/blob/4f6bad50ad5505010ffa2e39330d661a914826e7/test/integration/user_commands.cc#L141

https://github.com/gazebosim/gz-sim/blob/4f6bad50ad5505010ffa2e39330d661a914826e7/test/integration/user_commands.cc#L814

https://github.com/gazebosim/gz-sim/blob/4f6bad50ad5505010ffa2e39330d661a914826e7/test/integration/user_commands.cc#L1698

etc...

Could you clarify this more?